Add upgrade guide for the v27 release (#726)
humanmade/altis-documentation: other-docs/guides/upgrading/v27.md

- **CI moved to GitHub Actions (Developer Tools module)**
  Altis module CI has migrated from Travis CI to GitHub Actions. Each module repo now uses `.github/workflows/ci.yml`, a thin caller for the shared reusable workflow at `humanmade/altis-dev-tools/.github/workflows/module-ci.yml`.

  For projects you build on top of Altis, no action is required. Travis CI remains a fully supported option — your existing `.travis.yml` will continue to work. The skeleton no longer ships a Travis template by default, but you can keep maintaining one yourself; the `composer dev-tools phpunit` / `composer dev-tools codecept run` commands and the CI environment overrides documented in [Continuous Integration](docs://dev-tools/continuous-integration/) work identically under either CI.

  If you'd like to migrate your own project to GitHub Actions, the [Continuous Integration](docs://dev-tools/continuous-integration/) guide has a starter `ci.yml` template that wires up the new `altis-ci.yml` reusable workflow with DOCKER_USERNAME / DOCKER_PASSWORD secrets.

> - **[PHP 8.5](https://www.php.net/releases/8.5/en.php)** is now available, the new default version is `PHP 8.4` and the minimum required version is `PHP 8.2`. See [Updating PHP version](../updating-php/#altis-compatibility-chart) for the full compatibility chart.

Reviewer: Probably move this above the CI section.

### Altis Dashboard Features

> - The overhauled **Environment syncing** relies on `mydumper`/`myloader` for faster parallel streaming operation. You can select what to sync (database / uploads / both), as well as run post-sync actions (e.g. `search / replace`) after the sync is complete.

Reviewer: The fact we use mydumper/myloader is an internal thing, not appropriate here.
```suggestion
- The overhauled **Environment syncing** now uses faster parallel streaming to speed up the syncing. You can select what to sync (database / uploads / both), as well as run post-sync actions (e.g. `search / replace`) after the sync is complete.
```

> - The **Export** feature can also utilize the underlying `mydumper`/`myloader` tools. You can select what to export (database / uploads / both). The exported archive can then be imported into Altis Local Server.

Reviewer: ```suggestion
- The **Export** feature has also been enhanced. You can select what to export (database / uploads / both), and the exported archive can then be imported into Altis Local Server.
```

> - You now have the ability to **block requests by User-Agent** - useful for filtering requests, which match specific strings or patterns.

Reviewer: ```suggestion
- You now have the ability to **block requests by User-Agent** - useful for filtering requests which match specific strings or patterns.
```

- Some environment notifications are now dynamic (like an out of date Altis version), so they will self-clear once the prescribed action has been taken.
- As usual, we have updated dependencies to incorporate security patches and bug fixes

### Altis Local Server Improvements

> - Local Server now supports **[PHP 8.5](https://www.php.net/releases/8.5/en.php)**. Previous PHP versions are updated as per the [PHP compatibility chart](../updating-php/#altis-compatibility-chart).

Reviewer: Don't need this at all. We already mention we added PHP8.5

- Local server now supports **[MySQL 8.4](https://dev.mysql.com/doc/relnotes/mysql/8.4/en/)**. You can read a summary of what's changed in the [What Is New in MySQL 8.4 since MySQL 8.0](https://dev.mysql.com/doc/refman/8.4/en/mysql-nutshell.html) guide.
- If you have added your own [custom containers](https://docs.altis-dxp.com/local-server/extra-containers/) to local server, we now support **wildcard host matching**.

> - You can now **import** a full or partial export (database / uploads / both) from an Altis Environment in Local Server, using the `composer server import <file>` command. It is compatible with `mydumper`/`myloader` .

Reviewer: ```suggestion
- You can now **import** a full or partial export (database / uploads / both) from an Altis Environment in Local Server, using the `composer server import <file>` command.
```

```sh
# For cloud environments
wp altis migrate

# For local server
composer server cli -- altis migrate
```

## WordPress 7.0 "Armstrong"

  This release marks the beginning of a new era for WordPress — laying the foundation for AI across the platform, introducing a modernized admin experience, and shipping a broad set of new design and developer tools.

  - **WP AI Client** — A central hub for connecting AI providers (OpenAI, Anthropic, Google) directly to your site, managed via Settings > Connectors. Plugins can communicate with AI models through a unified interface without building their own integrations.
  - **Modernized Admin Dashboard** — A refreshed color palette, higher contrast, and upgraded typography across the entire admin, with smooth view transitions between screens (respecting reduced motion preferences).
  - **Command Palette** — Access any admin tool or screen instantly with ⌘K / Ctrl+K from anywhere in the dashboard.

For more details check the [WordPress 7.0 Field Guide](https://make.wordpress.org/core/2026/05/14/wordpress-7-0-field-guide/)

## Headline Features

> - **Improved environment syncing** 

Reviewer 2: Should we mention that this is a Dashboard feature as well?

Reviewer 2: OK. I see there is a dedicated Dashboard section, so this is for the CLI option/local server, right? Maybe clarify that.

  Environment syncing now implements faster stream copying between environments (e.g. prod to stage) with the option to run post-syncing commands (e.g. search/replace).  

- **CI moved to GitHub Actions (Developer Tools module)**

  Altis module CI has migrated from Travis CI to GitHub Actions. Each module repo now uses .github/workflows/ci.yml, a thin caller for the shared reusable workflow at humanmade/altis-dev-tools/.github/workflows/module-ci.yml.

>   For projects you build on top of Altis, no action is required. Travis CI remains a fully supported option — your existing .travis.yml will continue to work. The skeleton no longer ships a Travis template by default, but you can keep maintaining one yourself; the composer dev-tools phpunit / composer dev-tools codecept run commands and the CI environment overrides documented in Continuous Integration work identically under either CI.

Reviewer: Put the actual commands in back tics so they will be marked up as code in the rendered page.

>   If you'd like to migrate your own project to GitHub Actions, the Continuous Integration guide has a starter ci.yml template that wires up the new altis-ci.yml reusable workflow with DOCKER_USERNAME / DOCKER_PASSWORD secrets.

Reviewer 2: Add link to the Continuous Integration guide

>   The .travis.yml file originally created for a new Altis project is no longer created. Instead, a new install will create a file .github/workflows/ci.yml for you.

Reviewer: Don't really need this at all (most readers will be upgrading, not creating new).

> - **PHP 8.5** is now available, the new default version is `PHP 8.4`. See [Updating PHP version](../updating-php/#altis-compatibility-chart) for the full compatibility chart.

Reviewer: Maybe also link to https://www.php.net/releases/8.5/en.php

### Altis Dashboard Features

> - Improved environment syncing

Reviewer 2: On this section maybe a brief - one liner explanation on each improvement/new feature, what is has changed.

Reviewer: e.g. 
"We've improved the backup process by ...."
"X-ray now has feature 1 and feature 2..."
"Some environment notifications are now dynamic, (like eol warnings)_ so you no longer need to wait for us to clear those." Or just say, "Some notifications like an out of date Altis version are now self clearing."
"As usual, we have updated dependencies to incorporate security patches and bug fixes"

- Improved Backup
- Improvements in XRay
- Dynamic notifications
- Updated dependencies and bugfixes

### Altis Local Server Improvements

> - MySQL 8.4 support

Reviewer: PHP 8. 5 support is new
No need to mention the Traefik update (because we also updated the other images too), that should be covered by the normal "updated dependencies and libraries bit"
Private uploads is *not* added
The import-db deserves a full sentence and probably a link to the doc page.
The wildcard bit deserves a sentence starting "if you have added your own [custom containers](https://docs.altis-dxp.com/local-server/extra-containers/) to local server, we now support wildcard..."

- Replacing Redis container with Valkey
- Update Traefik image to latest
- Add GitHub Actions CI workflow
- Support private uploads
- Update PHP images and add PHP 8.5 support
- Use wildcard Host matchers instead of HostRegexp
- Add import-db command with mydumper/myloader support

### Breaking changes

- **Removed `altis-reusable-blocks` module**

> The package enhances WordPress's native reusable blocks (`wp_block` post type) with enterprise-level features. Given almost all the functionality is now available via the block editor, we are no longer going to install it by default. Users can install the plugin manually if they wish to continue using this feature.

Reviewer: Don't need the first sentence. Start with "Given almost all the functionality of this plugin is now..."
"we are no longer going to install it by default" > "we no longer install it by default"
Users > You
